pulp_2to3_migration/app/plugin/deb/migrator.py

```diff
@@ -0,0 +1,184 @@
+import asyncio
+
+from pulpcore.plugin.models import ProgressReport
+
+from pulpcore.plugin.stages import (
+    QueryExistingArtifacts,
+    ArtifactSaver,
+    QueryExistingContents,
+    ContentSaver,
+    ResolveContentFutures,
+    DeclarativeArtifact,
+    DeclarativeContent,
+)
+
+from pulp_deb.app import models as pulp3_models
+
+from pulp_2to3_migration.app.plugin.api import (
+    Pulp2to3PluginMigrator,
+    ContentMigrationFirstStage,
+    RelatePulp2to3Content,
+    DeclarativeContentMigration,
+)
+
+from pulp_2to3_migration.app.constants import (
+    DEFAULT_BATCH_SIZE,
+    NOT_USED,
+)
+
+from . import (
+    pulp2_models,
+    pulp_2to3_models,
+    repository,
+)
+
+
+class DebMigrator(Pulp2to3PluginMigrator):
+    """
+    An entry point for migration the Pulp 2 Debian plugin to Pulp 3.
+
+    Attributes:
+      Check parent class.
+    """
+    pulp2_plugin = 'deb'
+    pulp2_content_models = {
+        'deb': pulp2_models.DebPackage,
+    }
+    pulp2_collection = 'units_deb'
+    pulp3_plugin = 'pulp_deb'
+    pulp3_repository = pulp3_models.AptRepository
+    content_models = {
+        'deb': pulp_2to3_models.Pulp2DebPackage,
+    }
+    importer_migrators = {
+        'deb_importer': repository.DebImporter,
+    }
+    distributor_migrators = {
+        'deb_distributor': repository.DebDistributor,
+    }
+    future_types = {
+        'deb': pulp_2to3_models.Pulp2DebPackage,
+    }
+
+    @classmethod
+    def migrate_content_to_pulp3(cls):
+        """
+        Migrate pre-migrated Pulp 2 Debian content.
+        """
+        first_stage = DebContentMigrationFirstStage(cls)
+        dm = DebDeclarativeContentMigration(first_stage=first_stage)
+
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(dm.create())
+
+
+class DebDeclarativeContentMigration(DeclarativeContentMigration):
+    """
+    A pipeline that migrates pre-migrated Pulp 2 deb content into Pulp 3.
+    """
+
+    def pipeline_stages(self):
+        """
+        Build a list of stages.
+
+        This defines the "architecture" of the content migration to Pulp 3.
+
+        Returns:
+            list: List of :class:`~pulpcore.plugin.stages.Stage` instances
+
+        """
+        pipeline = [
+            self.first_stage,
+            QueryExistingArtifacts(),
+            ArtifactSaver(),
+            QueryExistingContents(),
+            ContentSaver(),
+            ResolveContentFutures(),
+            RelatePulp2to3Content(),
+        ]
+
+        return pipeline
+
+
+class DebContentMigrationFirstStage(ContentMigrationFirstStage):
```

Comment:
@quba42 , could you share what pulp_deb needs here that the default implementation doesn't cover? Why do you need  to override `migrate_to_pulp3`?

Reply:
Yes, actually I had meant to comment on that. I separated that out in order to better understand it (strip out the many if clauses that I was not actually using. However, at this point I don't think I am actually using anything I could not achieve with `ContentMigrationFirstStage`. I will retest to make sure, and remove this if it works without.

I may need to add it back in when I add the additional content types.

Reply:
It works without, so I dropped this subclassing. I also fixed a couple of linter errors.